modules/gdscript/gdscript_parser.cpp

```diff
@@ -3950,11 +3951,11 @@ bool GDScriptParser::export_annotations(const AnnotationNode *p_annotation, Node
 		}
 		hint_string += arg_string;
 	}
-
 	variable->export_info.hint_string = hint_string;
 
 	// This is called after the analyzer is done finding the type, so this should be set here.
 	DataType export_type = variable->get_datatype();
+	bool use_default_variable_type_check = true;
```

Comment:
Might be better to invert it so we explicitly opt-in annotations that require it. The old code was doing basically the same.

Reply:
Note that default variable type check must be done for all unlisted export annotations too, not just for `@export_range` and `@export_multiline`. What you are asking can be done, but it will require adding an `else` block. I think "default" is a pretty clear indication that the check below works unless explicitly disabled.

Reply:
Makes sense.